joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50
joint1_position_controller:
  type: position_controllers/JointPositionController
  joint: kinova_joint_1
  pid: {p: 100.0, i: 0.1, d: 10.0}
joint2_position_controller:
  type: position_controllers/JointPositionController
  joint: kinova_joint_2
  pid: {p: 100.0, i: 0.1, d: 10.0}
joint3_position_controller:
  type: position_controllers/JointPositionController
  joint: kinova_joint_3
  pid: {p: 100.0, i: 0.1, d: 10.0}
joint4_position_controller:
  type: position_controllers/JointPositionController
  joint: kinova_joint_4
  pid: {p: 100.0, i: 0.1, d: 10.0}
joint5_position_controller:
  type: position_controllers/JointPositionController
  joint: kinova_joint_5
  pid: {p: 100.0, i: 0.1, d: 10.0}
joint6_position_controller:
  type: position_controllers/JointPositionController
  joint: kinova_joint_6
  pid: {p: 100.0, i: 0.1, d: 10.0}
joint7_position_controller:
  type: position_controllers/JointPositionController
  joint: kinova_joint_7
  pid: {p: 100.0, i: 0.1, d: 10.0}
finger_position_controller:
  type: position_controllers/JointPositionController
  joint: finger_joint
  pid: {p: 10.0, i: 0.1, d: 1.0}
finger2_position_controller:
  type: position_controllers/JointPositionController
  joint: right_outer_knuckle_joint
  pid: {p: 10.0, i: 0.1, d: 1.0}
# controller_list:
#   - name: arm_controller
#     action_ns: follow_joint_trajectory
#     default: True
#     type: FollowJointTrajectory
#     joints:
#       - kinova_joint_1
#       - kinova_joint_2
#       - kinova_joint_3
#       - kinova_joint_4
#       - kinova_joint_5
#       - kinova_joint_6
#       - kinova_joint_7
#   - name: end_effector_controller
#     action_ns: follow_joint_trajectory
#     default: True
#     type: FollowJointTrajectory
#     joints:
#       - finger_joint
# arm_controller:
#   type: position_controllers/JointTrajectoryController
#   joints:
#     - kinova_joint_1
#     - kinova_joint_2
#     - kinova_joint_3
#     - kinova_joint_4
#     - kinova_joint_5
#     - kinova_joint_6
#     - kinova_joint_7
#   constraints:
#     goal_time: 0.6
#     stopped_velocity_tolerance: 0
#     kinova_joint_1: {trajectory: 0.1, goal: 0.01}
#     kinova_joint_2: {trajectory: 0.1, goal: 0.01}
#     kinova_joint_3: {trajectory: 0.1, goal: 0.01}
#     kinova_joint_4: {trajectory: 0.1, goal: 0.01}
#     kinova_joint_5: {trajectory: 0.1, goal: 0.01}
#     kinova_joint_6: {trajectory: 0.1, goal: 0.01}
#     kinova_joint_7: {trajectory: 0.1, goal: 0.01}
#   gains:
#     kinova_joint_1:
#       p: 1000
#       d: 50
#       i: 10
#       i_clamp: 1
#     kinova_joint_2:
#       p: 5000
#       d: 30
#       i: 100
#       i_clamp: 4
#     kinova_joint_3:
#       p: 5000
#       d: 50
#       i: 10
#       i_clamp: 1
#     kinova_joint_4:
#       p: 200
#       d: 20
#       i: 10
#       i_clamp: 1
#     kinova_joint_5:
#       p: 100
#       d: 10
#       i: 0.1
#       i_clamp: 1
#     kinova_joint_6:
#       p: 10
#       d: 1
#       i: 0.1
#       i_clamp: 1
#     kinova_joint_7:
#       p: 10
#       d: 1
#       i: 0.1
#       i_clamp: 1
# end_effector_controller:
#   type: position_controllers/JointTrajectoryController
#   joints:
#     - finger_joint
#   gains:
#     robotiq_85_left_knuckle_joint:
#       p: 80
#       d: 0
#       i: 0
#       i_clamp: 1